tests/stress/km/stress_tests_km.cpp

```diff
@@ -1084,6 +1091,296 @@ _print_test_control_info(const test_control_info& test_control_info)
+    if (prog_map_fd < 0) {
+        LOG_ERROR(
+            "({}) FATAL ERROR: bpf_object__find_map_fd_by_name({}) failed. Errno:{}",
+            __func__,
+            map_name.c_str(),
+            errno);
+        REQUIRE(prog_map_fd >= 0);
+    }
+    LOG_VERBOSE("({}) Opened fd:{} for map:{}", __func__, prog_map_fd, map_name.c_str());
+
+    // Set up tail calls.
+    for (int index = 0; index < MAX_TAIL_CALL_CNT; index++) {
+        try {
+            std::string bind_program_name{"BindMonitor_Callee"};
+            bind_program_name += std::to_string(index);
+
+            // Try to get a handle to the 'BindMonitor_Callee<index>' program.
+            bpf_program* callee = bpf_object__find_program_by_name(object, bind_program_name.c_str());
+
+            if (callee == nullptr) {
+                LOG_VERBOSE("({}) - bpf_object__find_program_by_name() failed. errno: {}", bind_program_name, errno);
+                REQUIRE(callee != nullptr);
+            }
+            LOG_VERBOSE("({}) - bpf_object__find_program_by_name() success.", bind_program_name);
+
+            fd_t callee_fd = bpf_program__fd(callee);
+            if (callee_fd < 0) {
+                LOG_VERBOSE("({}) - bpf_program__fd() failed. errno: {}", bind_program_name, errno);
+                REQUIRE(callee_fd > 0);
+            }
+            LOG_VERBOSE("({}) - bpf_program__fd() for callee_fd:{} success.", bind_program_name, callee_fd);
+
+            uint32_t result = bpf_map_update_elem(prog_map_fd, &index, &callee_fd, 0);
+            if (result < 0) {
+                LOG_VERBOSE("({}) - bpf_map_update_elem() failed. errno: {}", bind_program_name, errno);
+                REQUIRE(result == ERROR_SUCCESS);
+            }
+            LOG_VERBOSE("({}) - bpf_map_update_elem() for callee_fd:{} success.", bind_program_name, callee_fd);
+
+        } catch (...) {
+            // No need to terminate. We don't care about user mode issues here.
+        }
+    }
+}
+
+static void
+_invoke_mt_bindmonitor_tail_call_thread_function(thread_context& context)
+{
+    // Test bind.
+    SOCKET socket_handle = INVALID_SOCKET;
+    SOCKADDR_STORAGE remote_endpoint{};
+
+    if (context.role == thread_role_type::MONITOR_IPV4) {
+        remote_endpoint.ss_family = AF_INET;
+    } else {
+        ASSERT(context.role == thread_role_type::MONITOR_IPV6);
+        remote_endpoint.ss_family = AF_INET6;
+    }
+
+    uint16_t remote_port = SOCKET_TEST_PORT + static_cast<uint16_t>(context.thread_index);
+    using sc = std::chrono::steady_clock;
+    auto endtime = sc::now() + std::chrono::minutes(context.duration_minutes);
+    while (sc::now() < endtime) {
+
+        // Now send out a small burst of TCP 'bind' attempts for the duration of the test.  We do this to increase
+        // the probability of a collision between the program invocation and the extension being restarted at the same
+        // time.
+        constexpr uint32_t BURST_SIZE = 4;
```

Comment:
Just curious why the burst size value was set to `4`?  At first blush, this call is localized to the local stack does not result in any (call blocking) network traffic, so a larger value would have been fine (and seemingly preferrable) as well?

Reply:
> The number 4 was selected taking reference from 'connect' call in the test case.

I'd recommend caution while using constant values from other (unrelated) locations in this source file.

The `connect()` call location (with the BURST_SIZE of `4`) alluded to in your response is also preceded by a comment giving the specific reason for the small burst size, with additional details in my prior comment.  FWIW, there's also another `connect()` call usage in this source file that uses a burst size of `8192`.

In summary, the BURST_SIZE constant value is determined by the use case and in this specific instance, the rationale for the smaller value does not apply. Unless there exists a more pressing reason for staying with the current value, I'd recommend bumping this up (4K or 8K) modulo expected test thread runtime.  This will ensure the maximum possible stress on the 'invoke+tailcall' code path on a _per-thread_ basis as well.

Reply:
Re. the last paragraph in my prior comment, on further thought the smaller burst size will be compensated by a higher number of the 'execution time' bounded iterations, so we'll even out in the end insofar as per-thread stress load is concerned.  A burst size of 4 should therefore be fine.   Or we could simplify this further and get rid of the inner, burst size loop completely as it's not really buying us anything. I'm good with either approach.

Reply:
Thanks. Just made it 5, and kept the same logic.